### PR TITLE
update the translation of `Array.push()`

### PR DESCRIPTION
**Pull request by robert-jx**
### Description

update the translation of array.push()
### Motivation

update the translation of array.push()
### Additional details
@JasonLamv-t

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md [hunk 1]
```diff
@@ -20,78 +21,96 @@ push(element0, element1, /* … ,*/ elementN)
...
+// 这里没有长度属性，所以长度为 0
+Array.prototype.push.call(plainObj, 1, 2);
+console.log(plainObj);
+// { '0': 1, '1': 2, length: 2 }
 ```
 
-### 像数组一样使用对象
+### 以类似数组的方式使用对象
+
+如上所述，`push` 是通用的，我们可以利用这一点。就像这个例子所展示的，`Array.prototype.push` 可以在一个对象中正常运作。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/push/index.md:93]:
```suggestion
如上所述，`push` 被有意设计为通用的，我们可以利用这一点。就像这个例子所展示的，`Array.prototype.push` 可以很好的处理对象。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md [hunk 2]
```diff
@@ -20,78 +21,96 @@ push(element0, element1, /* … ,*/ elementN)
...
 
-如上所述，push 是特意设计为通用的，我们可以使用它来获得便利。正如下面的例子所示，Array.prototype.push 可以在一个对象上工作。注意，我们没有创建一个数组来存储对象的集合。相反，我们将该集合存储在对象本身上，并使用在 Array.prototype.push 上使用的 `call` 来调用该方法，使其认为我们正在处理数组，而它只是像平常一样运作，这要感谢 JavaScript 允许我们建立任意的执行上下文。
+注意，我们没有创建一个数组来存储对象集合。相反，我们将集合存储在对象本身上，并使用`Array.prototype.push` 的 `call` 来欺骗方法，让它认为我们正在处理一个数组——它只是工作，感谢JavaScript允许我们以任何我们想要的方式建立执行上下文的方式。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/push/index.md:95]:
```suggestion
注意，我们没有创建一个数组来存储对象集合。相反，我们将集合存储在对象本身上，并通过`Array.prototype.push` 的 `call` 来调用该方法，让它认为我们正在处理一个数组——归功于 JavaScript 允许我们以任何我们想要的方式建立执行上下文的方式，这样是可行的。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md [hunk 3]
```diff
@@ -20,78 +21,96 @@ push(element0, element1, /* … ,*/ elementN)
...
+// 让我们添加一些空对象来说明
 obj.addElem({});
 obj.addElem({});
-console.log(obj.length);
-// → 2
+console.log(obj.length); // 2
 ```
 
-注意，尽管 obj 不是数组，但是 push 方法成功地使 obj 的 length 属性增长了，就像我们处理一个实际的数组一样。
+注意，尽管 `obj` 不是一个数组，`push` 方法仍像处理实际数组那般成功地增加了 `obj` 的 `length` 属性。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/push/index.md:113]:
```suggestion
注意，尽管 `obj` 不是一个数组，`push` 方法仍成功地增加了 `obj` 的 `length` 属性，就像我们处理一个真正的数组一样。
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md [hunk 4]
```diff
@@ -20,78 +21,96 @@ push(element0, element1, /* … ,*/ elementN)
...
 ## 描述
 
-push 方法将值追加到数组中。
+`push()` 方法将值追加到一个数组中。
 
-`push` 方法具有通用性。该方法和 {{jsxref("Function.call", "call()")}} 或 {{jsxref("Function.apply", "apply()")}} 一起使用时，可应用在类似数组的对象上。`push` 方法根据 `length` 属性来决定从哪里开始插入给定的值。如果 `length` 不能被转成一个数值，则插入的元素索引为 0，包括 `length` 不存在时。当 `length` 不存在时，将会创建它。
+{{jsxref("Array.prototype.unshift()")}} 有着和 `push()` 相似的行为，但是其作用于数组的开头。
 
-唯一的原生类数组（array-like）对象是 {{jsxref("Global_Objects/String", "Strings")}}，尽管如此，它们并不适用该方法，因为字符串是不可改变的。
+`push()` 方法是一个改变方法。它改变了 `this` 的内容和长度。如果你希望 `this` 的值保持不变，但返回一个末尾追加了元素的新数组，你可以使用 [`arr.concat([element0, element1, /* ... ,*/ elementN])`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) 来代替。请注意，这些元素需要被包装在一个额外的数组中——否则，如果元素本身是一个数组，则由于 `concat()` 的行为，它将被展开且其而不是作为单个元素添加到原数组的末尾。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/push/index.md:36]:
```suggestion
`push()` 方法是一个修改方法。它改变了 `this` 的内容和长度。如果你希望 `this` 的值保持不变，但返回一个末尾追加了元素的新数组，你可以使用 [`arr.concat([element0, element1, /* ... ,*/ elementN])`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) 来代替。请注意，这些元素需要被包装在一个额外的数组中——否则，如果元素本身是一个数组，则由于 `concat()` 的行为，它将被展开且其而不是作为单个元素添加到原数组的末尾。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md [hunk 5]
```diff
@@ -20,78 +21,96 @@ push(element0, element1, /* … ,*/ elementN)
...
 ## 描述
 
-push 方法将值追加到数组中。
+`push()` 方法将值追加到一个数组中。
 
-`push` 方法具有通用性。该方法和 {{jsxref("Function.call", "call()")}} 或 {{jsxref("Function.apply", "apply()")}} 一起使用时，可应用在类似数组的对象上。`push` 方法根据 `length` 属性来决定从哪里开始插入给定的值。如果 `length` 不能被转成一个数值，则插入的元素索引为 0，包括 `length` 不存在时。当 `length` 不存在时，将会创建它。
+{{jsxref("Array.prototype.unshift()")}} 有着和 `push()` 相似的行为，但是其作用于数组的开头。
 
-唯一的原生类数组（array-like）对象是 {{jsxref("Global_Objects/String", "Strings")}}，尽管如此，它们并不适用该方法，因为字符串是不可改变的。
+`push()` 方法是一个修改方法。它改变了 `this` 的内容和长度。如果你希望 `this` 的值保持不变，但返回一个末尾追加了元素的新数组，你可以使用 [`arr.concat([element0, element1, /* ... ,*/ elementN])`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) 来代替。请注意，这些元素需要被包装在一个额外的数组中——否则，如果元素本身是一个数组，则由于 `concat()` 的行为，它将被展开且其而不是作为单个元素添加到原数组的末尾。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/push/index.md:36]:
```suggestion
`push()` 方法是一个修改方法。它改变了 `this` 的内容和长度。如果你希望 `this` 的值保持不变，但返回一个末尾追加了元素的新数组，你可以使用 [`arr.concat([element0, element1, /* ... ,*/ elementN])`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) 来代替。请注意，这些元素需要被包装在一个额外的数组中——否则，如果元素本身是一个数组，由于 `concat()` 的行为，它将被展开且而不是作为单个元素添加到原数组的末尾。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md [hunk 6]
```diff
@@ -20,78 +21,96 @@ push(element0, element1, /* … ,*/ elementN)
...
+
+`push()` 方法是[通用的](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#通用数组方法)。它只期望 `this` 值具有 `length` 属性和整数键属性。虽然字符串也是类似数组的，但这个方法不适合应用于它们，因为字符串是不可变的。
 
 ## 示例
 
-### 添加元素到数组
+### 在数组中添加元素
 
-下面的代码创建了 `sports` 数组，包含两个元素，然后又把两个元素添加给它。`total` 变量为数组的新长度值。
+接下来的代码中将会生成包含两个元素的 `sports` 数组，然后向其添加两个元素。`total` 变量将会包含新的数组长度。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/push/index.md:44]:
```suggestion
接下来的代码中将会生成包含两个元素的 `sports` 数组，然后向其添加两个元素。`total` 变量将会记录数组的新长度。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md [hunk 7]
```diff
@@ -33,15 +33,15 @@ push(element0, element1, /* … ,*/ elementN)
 
 {{jsxref("Array.prototype.unshift()")}} 有着和 `push()` 相似的行为，但是其作用于数组的开头。
 
-`push()` 方法是一个修改方法。它改变了 `this` 的内容和长度。如果你希望 `this` 的值保持不变，但返回一个末尾追加了元素的新数组，你可以使用 [`arr.concat([element0, element1, /* ... ,*/ elementN])`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) 来代替。请注意，这些元素需要被包装在一个额外的数组中——否则，如果元素本身是一个数组，则由于 `concat()` 的行为，它将被展开且其而不是作为单个元素添加到原数组的末尾。
+`push()` 方法是一个修改方法。它改变了 `this` 的内容和长度。如果你希望 `this` 的值保持不变，但返回一个末尾追加了元素的新数组，你可以使用 [`arr.concat([element0, element1, /* ... ,*/ elementN])`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) 来代替。请注意，这些元素需要被包装在一个额外的数组中——否则，如果元素本身是一个数组，由于 `concat()` 的行为，它将被展开且而不是作为单个元素添加到原数组的末尾。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/push/index.md:36]:
```suggestion
`push()` 方法是一个修改方法。它改变了 `this` 的内容和长度。如果你希望 `this` 的值保持不变，但返回一个末尾追加了元素的新数组，你可以使用 [`arr.concat([element0, element1, /* ... ,*/ elementN])`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) 来代替。请注意，这些元素需要被包装在一个额外的数组中——否则，如果元素本身是一个数组，由于 `concat()` 的行为，它将被展开而不是作为单个元素添加到原数组的末尾。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md [hunk 8]
```diff
@@ -20,78 +21,96 @@ push(element0, element1, /* … ,*/ elementN)
...
+console.log(sports); // ['soccer', 'baseball', 'football', 'swimming']
+console.log(total); // 4
 ```
 
 ### 合并两个数组
 
-该示例使用 {{jsxref("Function.apply", "apply()")}} 添加第二个数组的所有元素。
-
-注意当第二个数组 (如示例中的 moreVegs) 太大时不要使用这个方法来合并数组，因为事实上一个函数能够接受的参数个数是有限制的。具体可以参考 {{jsxref("Function.apply", "apply()")}} 。
+这个示例使用 {{jsxref("Operators/Spread_syntax", "spread syntax", "", "1")}} 来将第二个数组的所有元素添加到第一个数组中。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/push/index.md:56]:
```suggestion
这个示例使用{{jsxref("Operators/Spread_syntax", "展开语法", "", "1")}}来将第二个数组的所有元素添加到第一个数组中。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md [hunk 9]
```diff
@@ -20,78 +21,96 @@ push(element0, element1, /* … ,*/ elementN)
...
-push 方法将值追加到数组中。
+`push()` 方法将值追加到一个数组中。
 
-`push` 方法具有通用性。该方法和 {{jsxref("Function.call", "call()")}} 或 {{jsxref("Function.apply", "apply()")}} 一起使用时，可应用在类似数组的对象上。`push` 方法根据 `length` 属性来决定从哪里开始插入给定的值。如果 `length` 不能被转成一个数值，则插入的元素索引为 0，包括 `length` 不存在时。当 `length` 不存在时，将会创建它。
+{{jsxref("Array.prototype.unshift()")}} 有着和 `push()` 相似的行为，但是其作用于数组的开头。
 
-唯一的原生类数组（array-like）对象是 {{jsxref("Global_Objects/String", "Strings")}}，尽管如此，它们并不适用该方法，因为字符串是不可改变的。
+`push()` 方法是一个修改方法。它改变了 `this` 的内容和长度。如果你希望 `this` 的值保持不变，但返回一个末尾追加了元素的新数组，你可以使用 [`arr.concat([element0, element1, /* ... ,*/ elementN])`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) 来代替。请注意，这些元素需要被包装在一个额外的数组中——否则，如果元素本身是一个数组，由于 `concat()` 的行为，它将被展开而不是作为单个元素添加到原数组的末尾。
+
+`push()` 方法是[通用的](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#通用数组方法)。它只期望 `this` 值具有 `length` 属性和整数键属性。虽然字符串也是类似数组的，但这个方法不适用于它们，因为字符串是不可变的。
```
[the remark this answers is not in the document]
Comment (robert-jx) [reply files/zh-cn/web/javascript/reference/global_objects/array/push/index.md:38]: 嗯嗯好的👌

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md (hunk 8 above)
Comment (robert-jx) [reply files/zh-cn/web/javascript/reference/global_objects/array/push/index.md:56]: 有个问题，像这种里面的要怎么判断要不要翻译它 {{jsxref("Operators/Spread_syntax", "spread syntax", "", "1")}}🫡

Comment (yin1999) [reply files/zh-cn/web/javascript/reference/global_objects/array/push/index.md]: 参考 https://developer.mozilla.org/zh-CN/docs/MDN/Writing_guidelines/Page_structures/Macros/Commonly_used_macros

了解一下每一个参数的作用

**Review by yin1999 — APPROVED** [review]

File: files/zh-cn/web/javascript/reference/global_objects/array/push/index.md (hunk 8 above)
Comment (robert-jx) [reply files/zh-cn/web/javascript/reference/global_objects/array/push/index.md]: 嗯嗯好的

**Pull request merged**